lib/net/ldap/connection.rb

```diff
@@ -691,10 +727,19 @@ def delete(args)
     write(request, controls, message_id)
     pdu = queued_read(message_id)
 
-    if !pdu || pdu.app_tag != Net::LDAP::PDU::DeleteResponse
-      raise Net::LDAP::LdapError, "response missing or invalid"
-    end
+    assert_response!(pdu, Net::LDAP::PDU::DeleteResponse)
 
     pdu
   end
+
+  ##
+  # Raise +Net::LDAP::LdapError+ if the response is missing or an unexpected
+  # response tag is returned.
+
+  def assert_response!(pdu, tag)
+    return if pdu && pdu.app_tag == tag
+    raise Net::LDAP::LdapError, 'response missing or invalid'
+  end
+  private :assert_response!
```

Comment:
:+1: to this abstraction.

Couple of nitpicks:

* What's your reasoning for adding the `!` in `assert_response!`?
* [Minitest assertions](http://ruby-doc.org/stdlib-2.0/libdoc/minitest/rdoc/MiniTest/Assertions.html#method-i-assert_equal) typically have the expected value as the first argument, and the subject under test as the 2nd. It'd be good to flip the order of your argument list to match unless there's a good reason otherwise.
* Rather than raise an exception, you can use [`flunk`](http://ruby-doc.org/stdlib-2.0/libdoc/minitest/rdoc/MiniTest/Assertions.html#method-i-flunk) with a descriptive message. Something like:

```ruby
unless pdu && pdu.app_tag == tag
  flunk "Expected #{tag.inspect}, but got #{(pdu && pdu.app_tag).inspect}"
end
```